cli/src/pcluster/models/cluster_resources.py

```diff
@@ -183,6 +188,24 @@ class ClusterCreationFailure:
             )
         return failure.failure_code, failure.api_failure_reason
 
+    def _get_alarm_names(self):
+        """Get alarm names using cw stack resources."""
+        stack_resources = describe_stack_resources(self.name)
+        alarm_names = []
+        for resource in stack_resources.values():
+            if resource.get("ResourceType") == "AWS::CloudWatch::Alarm":
+                alarm_name = resource.get("PhysicalResourceId")
+                if alarm_name:
+                    alarm_names.append(alarm_name)
+        return alarm_names
```

Comment:
nit: this can be simplified as:
```
return [resource.get("PhysicalResourceId") for resource in stack_resources.values()
            if resource.get("ResourceType") == "AWS::CloudWatch::Alarm" and resource.get("PhysicalResourceId")]
```

Reply:
Thanks! Revised as you suggested.